Merge v0.1.4 for release

amoshydra opened the pull request.

tayjiehao committed:
Change input parameter of getFileByID() and getFolderByID() to String

tayjiehao committed:
Add a check for illegal characters in rename()

chuajiaxuan committed:
Debug on vagrant instead of travis

chuajiaxuan committed:
Debug text difference error

chuajiaxuan committed:
Update test case

chuajiaxuan committed:
Resolved Test issue with Travis

chuajiaxuan committed:
Add comments for diagramRenderer spec

chuajiaxuan committed:
Add comments explaining test case

chuajiaxuan committed:
Fix uncaught errors

chuajiaxuan committed:
Add new test case to handle invalid syntax

chuajiaxuan committed:
Add XSS filtering on diagram content

jiayingy committed:
Handle editor, viewer and split screen widths with css instead of dynamic classes

jiayingy committed:
Remove hover-grow effect on buttons

jiayingy committed:
Add logo with text

jiayingy committed:
Replace sashimi note logo

jiayingy committed:
Edit sashimi note logo for font to remain the same across all browsers

chuajiaxuan committed:
Fix test case bug

chuajiaxuan committed:
Remove vagrant from repo

chuajiaxuan committed:
Refactor test file codes

chuajiaxuan committed:
Attempt to merge dev/diagramsRenderer to v0.1.4

jiayingy committed:
Add icons to assets

chuajiaxuan committed:
Discover flowchart bug on pages and slides

chuajiaxuan committed:
Discover bug drawing flowchart in pages and slides

jiayingy committed:
Handle pasting of rich text to file/folder name will return plain text

jiayingy committed:
Fix broken editor and viewer view on smaller screen sizes

amoshydra committed:
Upgrade vue-codemirror and make it compatible

amoshydra committed:
Experiment with iframe as render target for HTML.vue

amoshydra committed:
Migrate styling into the iframe component

jiayingy committed:
Fix editor and viewer view on mobile

jiayingy committed:
Fix editor/viewer buttons on mobile

jiayingy committed:
Fix split screen button not appearing on desktop view

jiayingy committed:
Refactor search function

jiayingy committed:
Add wrapper class to file and folder

amoshydra committed:
Fix katex bug from missing doctype

amoshydra committed:
Change pageRenderer constructor to accept both id and element ref

amoshydra committed:
Add elementUtils for resolving element reference

tayjiehao committed:
Add check for deleted folders

tayjiehao committed:
Add copy function

amoshydra committed:
Add additional test case for elementUtils.spec

jiayingy committed:
Merge pull request #332 from nus-mtp/dev/fix-problem-with-codemirror

Fix problem with codemirror

amoshydra committed:
Make use of elementUtils resolve element

amoshydra committed:
Have documentNavigator to use the new element resolver

tayjiehao committed:
Fix incorrect variable names

tayjiehao committed:
Update test case for History

chuajiaxuan committed:
Fix bugs for pages and slides view

tayjiehao committed:
Update promise in rename()

amoshydra committed:
Merge pull request #335 from nus-mtp/dev/filemanager-v4

Dev/filemanager v4

chuajiaxuan committed:
Refactor diagramsRenderer.spec.js

chuajiaxuan committed:
Merge diagramsRenderer branch into current branch

tharain committed:
Merge branch 'refs/heads/database_v.1.2' into database_v.1.3

# Conflicts:
#	sashimi-webapp/src/database/data-modifier/dataAdd.js

tharain committed:
Remove function replaceAll in stringManipulator.js

not used anymore

tharain committed:
Add resolveSQLInjections function into stringManipulation.js

similar to replaceAll, but is used specifically to resolve the
problematic substring

tharain committed:
Add RevertSQLInjections in stringManipulation.js

To recover the resolveSQLInjection

tharain committed:
Add implementation of resolveSQLInjection into sqlCommands.js

tharain committed:
Add implementation of throwing an error when invalid character is typed in renaming in sqlCommands.js

tharain committed:
Add new exception to throw InvalidRename in sqlCommands.js

tharain committed:
Add implementation of revertSQLInjection in query.js

chuajiaxuan committed:
Debug on vagrant

chuajiaxuan committed:
Increase timeout for test case

jiayingy committed:
Add search and clear search icon to searchbar

tharain committed:
Fix unable to revert resolving of SQLinjection

[1] Revert back code for replaceAll
[2] Revert 2 resolve strings one by one

jiayingy committed:
Implement clearing of search string on searchbar

jiayingy committed:
Edit style for icon view

jiayingy committed:
Add width of searchbar icons

chuajiaxuan committed:
Debugging Test case

chuajiaxuan committed:
Fix debug error

chuajiaxuan committed:
Fix another test case error

chuajiaxuan committed:
Fix test error

chuajiaxuan committed:
Increasee timeout for some test case

tayjiehao committed:
Fix getFolderByID() and getFileByID() to allow both number and string input

jiayingy committed:
Update folder path when changing directory

tayjiehao committed:
Merge pull request #337 from nus-mtp/dev/filemanager-v4-bugfix

Bugfix for core component

tharain committed:
module exports of exception in facade class Storage.js

jiayingy committed:
Clear search input when changing directory

tharain committed:
Fix bug to initialize fileID to 1 instead of 0

tayjiehao committed:
Fix incorrect checking of File instance in rename()

tayjiehao committed:
Add download() function for File

tayjiehao committed:
Include creation date and last modified date for File and Folder

tharain committed:
Add check datatype is string in resolveSQLInjections in stringManipulation.js

tharain committed:
Fix deleteTable function in sqlCommands.spec.js

jiayingy committed:
Fix delete button when disabled will not delete files

tharain committed:
Add resolvingSQLInjection implementation into sqlCommands.js

tharain committed:
Remove redundant dependency dataDelete in entitiesCreator.spec.js

tharain committed:
Fix the indentation in sqlCommands.js

tayjiehao committed:
Fix download() to return promise

tharain committed:
Remove not working function deleteTable in sqlCommands.js

tharain committed:
Update implementation to createTable and insertContent test case for sqlCommands

tharain committed:
Update closedatabase and tablename in sqlCommands test case

tharain committed:
Add query testcase to search for path, partial search file name and getFullTable in sqlCommands test

tharain committed:
Merge remote-tracking branch 'refs/remotes/origin/development/v0.1.4' into database_v.1.3

amoshydra committed:
Merge pull request #342 from nus-mtp/dev/filemanager-v5

Dev/filemanager v5

tayjiehao committed:
Fix illegal character checking in rename()

amoshydra committed:
Rewrite DocumentNavigator to support iFrame

chuajiaxuan committed:
Include new API for conditionalProcessor.js

amoshydra committed:
Set touch-action programatically on the HTML element

amoshydra committed:
Fix problem where overwriteStyle fail if source is empty

amoshydra committed:
Enable better zooming result

1. When a page get flowed out to the left side of the screen, it will be
programatically brought back to the view
2. Zooming will retain scroll position on y-axis

amoshydra committed:
Fix compatibility issue with Firefox

amoshydra committed:
Bind all navigation event to the window

1. This is done because viewer will be utilising the iframe.
   The window object is constrained to the iframe only, and it
   doesn't npt affect other viewer target

amoshydra committed:
Fix empty renderDomTarget string id

amoshydra committed:
Upgrade PageRenderer to support iframe

amoshydra committed:
Update iframeBuilder to load stylesheet with Promise

amoshydra committed:
Refactor code to use this.html

amoshydra committed:
Remove hardcoded css styling by using js to style

amoshydra committed:
Remove unused beforeDestroy call

- Since the iframe will get removed upon unmount, the event attached to the content in iframe will get removed too.

amoshydra committed:
Contains all external markdown css into 1 file for import

amoshydra committed:
Update html.vue to use iframeBuilder

amoshydra committed:
Fix typo for unclosed style tag

amoshydra committed:
Merge pull request #344 from nus-mtp/dev/filemanager-v5.1

Fix illegal character checking in rename()

jiayingy committed:
Adjust logo to align centre vertically

jiayingy committed:
Remove unnecessary ai files

amoshydra committed:
Merge pull request #345 from nus-mtp/ui/fix-bugs

Merge UI bug fixes into development branch

chuajiaxuan committed:
Merge pull request #346 from nus-mtp/dev/encapsulate-viewer-with-iframe

Encapsulate viewer with iframe

chuajiaxuan committed:
Remove unnecessary comment

tharain committed:
Update syntax for done in entitiesCreator.js

tharain committed:
Remove unwanted import dataDelete in tableCreator.spec.js

tharain committed:
rename done in afterAll for tableCreator.spec.js

tharain committed:
Remove not working deleteTable function in sqlCommands.spec.js

Asynchronous will fail

tharain committed:
Update testcases for sqlCommands.spec.js

[1] shift before and after into its "describe"
[2] fix the asynchronous coding
[3] formulate the correct array for checking (alasql array for input and
returned array is different format)

tharain committed:
Change date format to long

Change all the implementations

chuajiaxuan committed:
Refactor code for markdown-it-diagram.js

chuajiaxuan committed:
Refactor code for diagramsRenderer

chuajiaxuan committed:
Fix possible issue for setting/clearing filename

chuajiaxuan committed:
Merge pull request #336 from nus-mtp/dev/merge-diagramsRenderer

Update development/v0.1.4 branch with diagram creation using textual representation

chuajiaxuan committed:
Resolve bug found when reviewing PR

amoshydra committed:
Merge pull request #348 from nus-mtp/database_v.1.3

Database v.1.3 - Fixing bugs

amoshydra committed:
Merge pull request #347 from nus-mtp/dev/conditional-rendererv2

Update development/v0.1.4 with new conditionalProcessor API

jiayingy committed:
Add margin below each folder/file button in icon view

jiayingy committed:
Fix alignment of navbar buttons

tayjiehao committed:
Fix regex for checking of illegal characters

jiayingy committed:
Add md-dark class to back and forward buttons

jiayingy committed:
Add margin below folder/file in icon view

jiayingy committed:
Set default view to icon view

jiayingy committed:
Set documents to list view on mobile view

jiayingy committed:
Fix button margins

jiayingy committed:
Reduce top margin of userActions

jiayingy committed:
Fix bug where icon does not represent the real state

jiayingy committed:
Call fileManager file download api

amoshydra committed:
Merge pull request #349 from nus-mtp/ui/fileManager-api-call

Call fileManager file download api

jiayingy committed:
Merge pull request #350 from nus-mtp/ui/enhance-fileManager-styling

Ui/enhance file manager styling

amoshydra committed:
Merge pull request #352 from nus-mtp/dev/filemanager-v5.1-bugfix

Fix regex for checking of illegal characters

chuajiaxuan committed:
Add ASCIIMath plugin to markdown-it

chuajiaxuan committed:
Add test case for inline asciiMath

chuajiaxuan committed:
Add test case for block asciiMath

amoshydra committed:
Fix slide and page not getting centralised

amoshydra committed:
Fix issue #306

amoshydra committed:
Commit babel-polyfill to resolve problem in IE

chuajiaxuan committed:
Add customised asciimath plugin library for webpack

chuajiaxuan committed:
Add the required ASCIIMath to Tex parser for webpack

chuajiaxuan committed:
Update markdownProcessor.js to use webpack configured asciimath plugin

chuajiaxuan committed:
Update asciiMathInlineOutput.txt

amoshydra committed:
Fix browser compatibility

amoshydra committed:
Merge pull request #354 from nus-mtp/dev/asciiMath-plugin

Add support for ASCIIMath in Markdown-it

tharain committed:
Add comments and checking for new function for dateTime.spec

tharain committed:
Fix issue #353

tayjiehao committed:
Merge pull request #355 from nus-mtp/fix/#306

Fix issue #306

tayjiehao committed:
Merge pull request #356 from nus-mtp/fix/slide-pages-layout

Fix slide and page not getting centralised

tayjiehao committed:
Merge pull request #357 from nus-mtp/fix/#305

Add babel-polyfill to resolve problem in IE

amoshydra committed:
Merge branch 'development/v0.1.4' into fix/browser-compatibility-for-document-navigator

amoshydra committed:
Merge pull request #358 from nus-mtp/fix/browser-compatibility-for-document-navigator

Fix browser compatibility for document navigator

amoshydra committed:
Change default fileName to empty string

amoshydra committed:
Implement condition switching

chuajiaxuan committed:
Fix asciiMath causing inline codes to be fenced code

amoshydra committed:
Merge pull request #362 from nus-mtp/dev/asciiMath-fix

Fix a bug caused by markdown-it-asciimath plugin

tharain committed:
Fix typo in dateTime.spec.js

amoshydra committed:
Remove debug console

chuajiaxuan committed:
Merge pull request #361 from nus-mtp/dev/ui-input-for-conditional-rendering

UI input for conditional rendering

amoshydra committed:
Merge pull request #359 from nus-mtp/database_v.1.3

Fix #353 with a added minor test case

amoshydra committed:
Merge branch 'master' into development/v0.1.4

jiayingy commented on sashimi-webapp/src/components/editor-viewer/Navbar.vue:
```diff
@@ -40,43 +37,71 @@
+              padding: 10px;
+              padding-left: 10px;
+              padding-right: 2px;"
+            />
```
Don't do inline styling. Do place them in the stylesheets

jiayingy commented on sashimi-webapp/src/components/editor-viewer/Viewers/floater/Floater.vue:
```diff
@@ -0,0 +1,56 @@
+    },
+    methods: {
+      goFullScreen() {
+        alert('what?');
```
this should not be pushed to master branch

jiayingy committed:
Show list view when screen size is < 480 on load

jiayingy committed:
Merge pull request #364 from nus-mtp/ui/fix-DocumentView-when-loaded

Show list view when screen size is < 480 on load

jiayingy committed:
Align logo to the left side for screen size < 768px

jiayingy committed:
Replace logo width on mobile view with variables

jiayingy committed:
Update variable name

jiayingy committed:
Merge pull request #365 from nus-mtp/ui/fix-logo-on-mobile-view

Ui/fix logo on mobile view

jiayingy committed:
Remove unnecessary line

jiayingy committed:
Remove styling of selection color

jiayingy committed:
Fix clearing of search string returns previous folder path user is at

jiayingy committed:
Use fileManager rename API

jiayingy committed:
Add class for folder and file name

jiayingy committed:
Style folder/file name when user is editing

amoshydra committed:
Make iframeBuilder addStyle to be compatible with more browser

amoshydra committed:
Handle css script loading error

amoshydra commented on sashimi-webapp/src/logic/documentPackager/markdown-it-asciimath.js:
```diff
@@ -0,0 +1,79 @@
+}
+
+function renderInline(str, disp) {
+  console.log(preprocessMath(str));
```
Unwanted console.log when rendering inline ascii math

amoshydra commented on sashimi-webapp/index.html:
```diff
@@ -28,6 +28,23 @@
+    <!-- Author of viz.js does not want to fix this issue, hence we are doing this -->
+    <!-- Author mention to use this unconventional/bad way of creating fake fs -->
+    <script>window.fs = {};</script>
+    <script src="./vendors/viz.js"></script>
```
These diagram rendering libraries (especially Mermaid and Viz) are taking huge amount of space (> 4mb). 
This caused a long loading time and big data consumption.

We should consider
- Load these library asynchronously https://developers.google.com/speed/docs/insights/BlockingJS
    - However this may not be feasible, as these library need to be loaded before the HTML content get rendered.
- Enable caching
    - This does not solve the initial loading time and load, however subsequent downloading will be drastically reduced.
- Enable gzipped compression on the server side

amoshydra commented on sashimi-webapp/index.html:
```diff
@@ -28,6 +28,23 @@
+    <script>window.fs = {};</script>
+    <script src="./vendors/viz.js"></script>
+    <!-- mermaid does not support markdown-it, hence we are doing this -->
+    <script src="./vendors/mermaidAPI.js"></script>
```
@chuajiaxuan Minified version of the libraries should be used instead. 
This goes for 
- sequence-diagram.js
- flowchart-latest.js
- viz.js (> 2mb)
- mermaidAPI.js (> 2mb)

chuajiaxuan replied:
Changed in PR #375! Viz.js is already minified according to the author. But i have used viz-lite.js instead as we don't seem to be using the packages in viz.js that are removed from viz-lite.js. The rest are changed to their respective minified versions.

amoshydra commented on sashimi-webapp/package.json:
```diff
@@ -68,6 +71,7 @@
     "lolex": "^1.5.2",
     "markdown-it": "^8.2.2",
     "markdown-it-anchor": "^3.0.0",
+    "markdown-it-asciimath": "^0.1.0",
```
@chuajiaxuan 
Remove npm dependency to `markdown-it-asciimath` since it is now~~t~~ copied as a customized in-house library.

chuajiaxuan replied:
Removed in PR #376!

amoshydra committed:
Adjust Pages and Slides paper size

amoshydra committed:
Update styling for Pages view

- Pages view now uses gutenberg oldstyle theme
- Code block for page view is not grey-scaled. Achieved by using "ascetic"

amoshydra committed:
Configure vscode for vetur vue

- Emmet is now usable in .vue
- Eslint will now highlight error automatically

jiayingy committed:
Merge pull request #369 from nus-mtp/setup/configure-vscode-for-vetur-vue

Configure vscode for vetur vue

jiayingy committed:
Merge branch 'development/v0.1.4' of https://github.com/nus-mtp/sashimi-note into development/v0.1.4

jiayingy committed:
Highlight 'icon' on viewType instead of 'list'

jiayingy committed:
Remove focus on name field when enter key is pressed

jiayingy committed:
Fix url path to display correct folder id when back or forward buttons are pressed

jiayingy committed:
Fix breadcrumb to show folder path if folder is accessed directly via url

chuajiaxuan committed:
Resolve ESLint warnings

jiayingy committed:
Resolve router-link warning

jiayingy committed:
Handle changing of folders when url query path is changed

jiayingy committed:
Remove router-link on current folder

jiayingy committed:
Remove link for current folder on breadcrumb

jiayingy committed:
Fix breadcrumb to show 'search results' instead of folder path

jiayingy committed:
Replace 'getElementsByTagName' with Vue's ref

tharain commented on sashimi-webapp/src/components/file-manager/Documents.vue:
```diff
@@ -117,29 +118,39 @@ export default {
+      // width: 60px;
+      // height: 40px;
+      // margin: auto;
+      // overflow: auto;
```
minor stuffs: I think when commenting out multiple lines it would be better to do the following instead of commenting out line by line

```
/*
position: absolute;
top: 15px;
left: 0;
bottom: 0;
right: 0;
width: 60px;
height: 40px;
margin: auto;
overflow: auto;
*/
```

tharain commented on sashimi-webapp/src/logic/filemanager/data/file.js:
```diff
@@ -136,23 +146,75 @@ File.prototype.move = function move(destFolder) {
+File.prototype.download = function download() {
+  return this.load()
+  .then((data) => {
+    const fileName = this.name.concat('.md');
```
we no longer have .md file names

amoshydra replied:
The `.md` file extension is used here, when the user download a file into their system.

- This is done so that the user can directly open the file with a markdown editor.
- It is also used to prevent user from downloading file that contain bad extension such as `.exe`

tharain replied:
ops ok paisei

tharain commented on sashimi-webapp/src/logic/documentPackager/ASCIIMathTeXImg.js:
```diff
@@ -0,0 +1,1073 @@
+    let h;
+    let m;
+    n = -1;
+    h = arr.length;
```
maybe can use more meaningful names, like `currIndex, nextIndex and arrLength` kind of confusing

amoshydra commented on sashimi-webapp/src/logic/filemanager/data/file.js:
```diff
@@ -136,23 +146,75 @@ File.prototype.move = function move(destFolder) {
+File.prototype.download = function download() {
+  return this.load()
+  .then((data) => {
+    const fileName = this.name.concat('.md');
```
The `.md` file extension is used here, when the user download a file into their system.

- This is done so that the user can directly open the file with a markdown editor.
- It is also used to prevent user from downloading file that contain bad extension such as `.exe`

chuajiaxuan committed:
Merge pull request #370 from nus-mtp/dev/asciiMath-fix

Resolve ESLint warnings

amoshydra committed:
Make use of iframeBuilder getDocument

amoshydra committed:
Add frameWindow validation code in getDocument

amoshydra committed:
Add test for iframeBuilder getDocument

amoshydra committed:
Remove unused iframeTestObj id

jiayingy committed:
Merge pull request #371 from nus-mtp/ui/fix-path-on-filemanager

Ui/fix path on filemanager

amoshydra committed:
Add new line at the end of file

amoshydra committed:
Add test for iframebuilder rebuild

amoshydra committed:
Add support of window object for iframeBuilder

amoshydra committed:
Rename iframeBuilder to documentBuilder

amoshydra committed:
Rename documentBuilder dependency

amoshydra committed:
Reimplement the print mechanism with javascript

amoshydra committed:
Seperate print logic into seperate file

amoshydra committed:
Support window object for elementUtils resolve

amoshydra committed:
Make use of resolveElement

amoshydra committed:
Implement CMD + P key for printing

amoshydra committed:
Make child iframe object responsive to print event

amoshydra committed:
Add newline at eof

amoshydra committed:
Update doc for elementUtils targetReference

amoshydra committed:
Update documentation on supporting Window object

amoshydra committed:
Merge pull request #367 from nus-mtp/fix/iframe-style-loader-run-indefinitely

Fix/iframe style loader run indefinitely

amoshydra committed:
Merge branch 'development/v0.1.4' into fix/#319-print-not-working

amoshydra committed:
Merge pull request #374 from nus-mtp/fix/#319-print-not-working

Fix #319 print not working

amoshydra committed:
Implement slide navigation functionality

amoshydra committed:
Combine repositioning algorithm for resize and zoom

amoshydra committed:
Remove left positioning code that does nothing

amoshydra committed:
Extract margin and transition var as global

amoshydra committed:
Make logic clearer with bracket

amoshydra committed:
Add margin to the left when zoomed

amoshydra committed:
Avoid event.ds to be evaluate to false when 0

amoshydra committed:
Fix miscalculated logic

chuajiaxuan committed:
Replace flowchart-latest.js with minified version

chuajiaxuan committed:
Update index.html to use minified flowchart.js

chuajiaxuan committed:
Replace mermaidAPI.js with minified version

chuajiaxuan committed:
Update index.html to use minified version of mermaidAPI

chuajiaxuan committed:
Remove unuse asciimath plugin

chuajiaxuan committed:
Replace sequence-diagram.js with minified version

chuajiaxuan committed:
Update index.html to use minified version of sequence-diagram.js

chuajiaxuan committed:
Replace viz.js with viz-lite.js

chuajiaxuan committed:
Update index.html to use viz-lite.js

amoshydra committed:
Merge branch 'development/v0.1.4' into ui/Restyle-pages-mode-with-gutenberg

amoshydra committed:
Merge pull request #368 from nus-mtp/ui/Restyle-pages-mode-with-gutenberg

Restyle pages mode with Gutenberg

jiayingy committed:
Merge pull request #375 from nus-mtp/dev/changeToMinified

Update development/v0.1.4 with new minified files

jiayingy committed:
Merge pull request #376 from nus-mtp/dev/remove-asciimathplugin

Remove asciimath plugin in from development/v0.1.4

tharain committed:
fix always saving file when throwing error

amoshydra committed:
Merge pull request #380 from nus-mtp/database_v.1.4

fix always saving file name even when throwing error

amoshydra committed:
Merge pull request #379 from nus-mtp/dev/refactor-document-navigator

Refactor document navigator

amoshydra committed:
Merge pull request #378 from nus-mtp/dev/slides-navigation-control

Implement slide navigation functionality

amoshydra committed:
Replace alert with console.log

This component is not being used right now. 
I have changed the alert box to console.log for now.

amoshydra committed:
Extract inline styling into the stylesheet

jiayingy committed:
Merge pull request #382 from nus-mtp/fix/styling-file-format-input

Extract inline styling into the stylesheet
